app/src/main/java/org/akvo/flow/util/logging/TagsFactory.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (C) 2017 Stichting Akvo (Akvo Foundation)
+ *
+ * This file is part of Akvo Flow.
+ *
+ * Akvo Flow is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Akvo Flow is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Akvo Flow.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package org.akvo.flow.util.logging;
+
+import android.content.Context;
+import android.content.pm.PackageInfo;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.support.annotation.NonNull;
+import android.text.TextUtils;
+
+import org.akvo.flow.util.ConstantUtil;
+import org.akvo.flow.util.PropertyUtil;
+import org.akvo.flow.util.StatusUtil;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import timber.log.Timber;
+
+public class TagsFactory {
+
+    private static final String INSTANCE_ID_TAG_KEY = "app.id";
+    private static final String DEVICE_ID_TAG_KEY = "flow.device.id";
+    private static final String DEVICE_MODEL_TAG_KEY = "device.model";
+    private static final String OS_VERSION_TAG_KEY = "os.version";
+    private static final String VERSION_NAME_TAG_KEY = "version.name";
+    private static final String VERSION_CODE_TAG_KEY = "version.code";
+    private static final String DEFAULT_TAG_VALUE = "NotSet";
+    private static final int NUMBER_OF_TAGS = 6;
+
+    private final Map<String, String> tags = new HashMap<String, String>(NUMBER_OF_TAGS);
```

Comment:
So in this case we should initialise it to 10 to avoid having the capacity increased, right?